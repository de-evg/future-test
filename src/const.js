export const showingStatus = {
    UNSET: `UNSET`,
    SHOW: `SHOW`,
    HIDE: `HIDE`
};

export const DEFAULT_SORT = `ID_UP`;

export const SortType = {
    ID_UP: `ID_UP`,
    ID_DOWN: `ID_DOWN`,
    FIRSTNAME_UP: `FIRSTNAME_UP`,
    FIRSTNAME_DOWN: `FIRSTNAME_DOWN`,
    LASTNAME_UP: `LASTNAME_UP`,
    LASTNAME_DOWN: `LASTNAME_DOWN`,
    EMAIL_UP: `EMAIL_UP`,
    EMAIL_DOWN: `EMAIL_DOWN`,
    PHONE_UP: `PHONE_UP`,
    PHONE_DOWN: `PHONE_DOWN`,
};