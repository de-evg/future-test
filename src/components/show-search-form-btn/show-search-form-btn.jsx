import React, { useCallback } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { ActionCreator } from "../../store/action";
import { NameSpace } from "../../store/reducers/root";
import { showingStatus } from "../../const";

const ShowSearchFormBtn = ({
  updateShowStatus,
  searchFormVisualStatus,
  users,
  isLoading,
}) => {
  const handleClick = useCallback(() => {
    updateShowStatus();
  }, [updateShowStatus]);
  const activeClass =
    searchFormVisualStatus === showingStatus.SHOW ? "button--active" : "";
  return (
    <button
      className={`main-menu__button button ${activeClass}`}
      onClick={handleClick}
      disabled={isLoading || !users.length}
    >
      Найти
    </button>
  );
};

ShowSearchFormBtn.propTypes = {
  updateShowStatus: PropTypes.func.isRequired,
  searchFormVisualStatus: PropTypes.string.isRequired,
  users: PropTypes.arrayOf(PropTypes.object).isRequired,
  isLoading: PropTypes.bool.isRequired,
};

const mapStateToProps = (state) => ({
  searchFormVisualStatus:
    state[NameSpace.INTERFACE_DATA].searchFormVisualStatus,
  users: state[NameSpace.USERS_DATA].users,
  isLoading: state[NameSpace.INTERFACE_DATA].isLoading,
});

const mapDispatchToProps = (dispatch) => ({
  updateShowStatus() {
    dispatch(ActionCreator.updateSearchFormStatus());
  },
});

export { ShowSearchFormBtn };
export default connect(mapStateToProps, mapDispatchToProps)(ShowSearchFormBtn);
