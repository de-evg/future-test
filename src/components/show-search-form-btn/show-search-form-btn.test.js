import React from "react";
import Enzyme, { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import { ShowSearchFormBtn } from "./show-search-form-btn";
import Adapter from "enzyme-adapter-react-16";
import {showingStatus} from "../../const";

Enzyme.configure({ adapter: new Adapter() });

describe(`Render ShowAddFormBtn correctly`, () => {
  it(`Reder ShowAddFormBtn correctly with UNSET showing status, TRUE isLoading`, () => {
    const wrapper = shallow(
      <ShowSearchFormBtn
        updateShowStatus={() => {}}
        searchFormVisualStatus={showingStatus.UNSET}
        isLoading={true}
        users={[{}, {}, {}]}
      />
    );
    const tree = shallowToJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it(`Reder ShowAddFormBtn correctly with SHOW showing status, FALSE isLoading`, () => {
    const wrapper = shallow(
      <ShowSearchFormBtn
        updateShowStatus={() => {}}
        searchFormVisualStatus={showingStatus.SHOW}
        isLoading={false}
        users={[{}, {}, {}]}
      />
    );
    const tree = shallowToJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it(`Reder ShowAddFormBtn correctly with HIDE showing status, FALSE isLoading`, () => {
    const wrapper = shallow(
      <ShowSearchFormBtn
        updateShowStatus={() => {}}
        searchFormVisualStatus={showingStatus.HIDE}
        isLoading={false}
        users={[{}, {}, {}]}
      />
    );
    const tree = shallowToJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});