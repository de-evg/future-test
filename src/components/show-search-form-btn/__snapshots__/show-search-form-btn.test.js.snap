// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ShowAddFormBtn correctly Reder ShowAddFormBtn correctly with HIDE showing status, FALSE isLoading 1`] = `
<button
  className="main-menu__button button "
  disabled={false}
  onClick={[Function]}
>
  Найти
</button>
`;

exports[`Render ShowAddFormBtn correctly Reder ShowAddFormBtn correctly with SHOW showing status, FALSE isLoading 1`] = `
<button
  className="main-menu__button button button--active"
  disabled={false}
  onClick={[Function]}
>
  Найти
</button>
`;

exports[`Render ShowAddFormBtn correctly Reder ShowAddFormBtn correctly with UNSET showing status, TRUE isLoading 1`] = `
<button
  className="main-menu__button button "
  disabled={true}
  onClick={[Function]}
>
  Найти
</button>
`;
